app/src/main/java/com/example/blah/mobilestudio/Activities/MainActivity.java

```diff
@@ -245,18 +277,53 @@ private void highlightFIle(String filePath) {
 
     private void resetBreadcrumb(String filePath) {
         //Horizontal Breadcrumb reset
-        breadFragment = (BreadcrumbFragment) getFragmentManager().findFragmentById(R.id.topBreadFragment);
+        breadFragment = (BreadcrumbFragment) getSupportFragmentManager().findFragmentById(R.id.topBreadFragment);
         breadFragment.currentPath = filePath;
 
         breadFragment.setOnClickListener(this);
-        getFragmentManager().beginTransaction().replace(R.id.topBreadFragment, breadFragment).commit();
+        getSupportFragmentManager()
+                .beginTransaction().
+                replace(R.id.topBreadFragment, breadFragment).commitAllowingStateLoss();
 
     }
 
     private void highlightBreadcrumbItem(String path) {
         ArrayList<String> items = new ArrayList(Arrays.asList(path.substring(1).split("\\s*/\\s*")));
-        breadFragment = (BreadcrumbFragment) getFragmentManager().findFragmentById(R.id.topBreadFragment);
+        breadFragment = (BreadcrumbFragment) getSupportFragmentManager().findFragmentById(R.id.topBreadFragment);
         breadFragment.highlightSelectedItem(items.get(items.size() - 1), items.size() - 1);
     }
 
```

Comment:
Can we put the following logic in file fragment codes instead of MainActivity? (If it's only for File Fragment)